### PR TITLE
Recalculate formula when dependant cell is set to "" 

### PR DESCRIPTION
**Pull request by pjanik**
https://www.pivotaltracker.com/story/show/186200846

This PR fixes a bug described in the PT story linked above. `if (c[attrId])` was not executed when `c[attrId]` was equal to `""`. I've refactored some helpers a bit and added unit tests to cover this use case.

### CODE REVIEW

File: v3/src/models/data/formula-utils.test.ts [hunk 1]
```diff
@@ -22,7 +24,7 @@ const displayNameMapExample: DisplayNameMap = {
     "Roller Coaster": {
       "id": "DATA_ROLLER_COASTER",
       "attribute": {
-        // \, ', and " are added to test characters escaping
+        // \, ", and " are added to test characters escaping
```
Comment (kswenson) [inline v3/src/models/data/formula-utils.test.ts:27]: Seems like this makes the comment redundant -- was this change intended?

**Review by kswenson — APPROVED** [review]
👍 Looks good

File: v3/src/models/data/formula-utils.test.ts (hunk 1 above)
Comment (pjanik) [reply v3/src/models/data/formula-utils.test.ts]: Thanks, it wasn't, probably some search-and-replace mistake. Fixed in the last commit.
